### ISSUE TITLE
Implement Storybook story for refactored version of WP dashboard widget

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2760]
#2075 and #2610 created a datastore-driven refactored version of the WP dashboard widget. However, this version is not yet covered in Storybook, there's only a story for the legacy version. Let's add a separate story for the new version. This should be done before merging #2240, to ensure we have the migration work completed before shipping it to end users.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be a new Storybook story for the _refactored_ version of the WP dashboard widget. It should be located in the same file as the legacy WP dashboard widget story.
* Except for the numbers used (we can use arbitrary mock data for the API requests), the new Storybook story should visually match the legacy one.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

* N/A

**Comment by ivankruchkoff** [comment]
I updated the storybook test in https://github.com/google/site-kit-wp/issues/2240 only need to extend it for #2610 work, so I'll update in that CR and will close out this ticket as the storybook story is being covered there.
